### PR TITLE
comparison added

### PR DESCRIPTION
**Pull request by AzizaAzhibekova**

### CODE REVIEW

File: src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import us.peaksoft.gadgetarium.enums.Subcat;
+import us.peaksoft.gadgetarium.exception.NotFoundException;
+import us.peaksoft.gadgetarium.repository.*;
+import us.peaksoft.gadgetarium.service.ComparisonService;
+
+@Service
+@RequiredArgsConstructor
+public class ComparisonServiceImpl implements ComparisonService {
+
+    private final CategoryRepository categoryRepository;
```
Comment (DilnazTolubaiKyzy) [inline src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java:20]: next line

File: src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    private final ProductRepository productRepository;
```
Comment (DilnazTolubaiKyzy) [inline src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java:21]: next line

File: src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    private final List<ComparisonResponse> smartphonesList = new ArrayList<>();
```
Comment (DilnazTolubaiKyzy) [inline src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java:22]: next line

File: src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    private final List<ComparisonResponse> tabletList = new ArrayList<>();
```
Comment (DilnazTolubaiKyzy) [inline src/main/java/us/peaksoft/gadgetarium/service/impl/ComparisonServiceImpl.java:23]: next line

**Review by DilnazTolubaiKyzy — COMMENTED** [review]
viewed

**Review by Aidai0404 — COMMENTED** [review]
viewed

File: src/main/java/us/peaksoft/gadgetarium/controller/ComparisonController.java [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+import io.swagger.v3.oas.annotations.tags.Tag;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import us.peaksoft.gadgetarium.dto.ComparisonResponse;
+import us.peaksoft.gadgetarium.enums.Subcat;
+import us.peaksoft.gadgetarium.service.ComparisonService;
+
+@RestController
+@RequestMapping("api/public/compare")
```
Comment (i-Coderr) [inline src/main/java/us/peaksoft/gadgetarium/controller/ComparisonController.java:14]: Pls remove public

**Review by i-Coderr — COMMENTED** [review]
Pls resolve all conversations

**Review by i-Coderr — APPROVED** [review]
Alright

**Pull request merged**